// @react-navigation/drawer/lib/commonjs/views/DrawerToggleButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DrawerToggleButton = DrawerToggleButton;
var _elements = require("@react-navigation/elements");
var _native = require("@react-navigation/native");
var _reactNative = require("react-native");
var _toggleDrawerIcon = _interopRequireDefault(require("./assets/toggle-drawer-icon.png"));
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
function DrawerToggleButton({
  tintColor,
  accessibilityLabel = 'Show navigation menu',
  imageSource = _toggleDrawerIcon.default,
  ...rest
}) {
  const navigation = (0, _native.useNavigation)();
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_elements.HeaderButton, {
    ...rest,
    accessibilityLabel: accessibilityLabel,
    onPress: () => navigation.dispatch(_native.DrawerActions.toggleDrawer()),
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactNative.Image, {
      resizeMode: "contain",
      source: imageSource,
      fadeDuration: 0,
      tintColor: tintColor,
      style: styles.icon
    })
  });
}
const styles = _reactNative.StyleSheet.create({
  icon: {
    height: 24,
    width: 24,
    marginVertical: 8,
    marginHorizontal: 5
  }
});
